py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,489 @@
+        """
+        Fetches an access/refresh token by authorizing OAuth 2.0 credentials.
+
+        Returns: True, if the authorization was successful.
+                 False, if a ServerNotFoundError is thrown.
+        """
+        try:
+            http = self.credentials.authorize(httplib2.Http())
+            self.service = discovery.build('calendar', 'v3', http=http)
+            return True
+        except ServerNotFoundError:
+            return False
+
+    def _get_events(self):
+        """
+        Fetches events from the color into a list.
+
+        Returns: The list of events.
+        """
+        self.last_update = datetime.datetime.now()
+        now = datetime.datetime.utcnow().isoformat() + 'Z'  # 'Z' indicates UTC time
+
+        event_cache = self.events
+
+        try:
+            eventsResult = self.service.events().list(
+                calendarId='primary', timeMin=now, maxResults=self.num_events, singleEvents=True,
+                orderBy='startTime').execute(num_retries=5)
+        except:
+            return event_cache
+
+        return eventsResult.get('items', [])
+
+    def _check_warn_threshold(self, time_to, start, end, summary):
+        """
+        Checks if the time until an event starts is less than or equal to the warn_threshold.
+        If it is, it presents a warning with self.py3.notify_user.
+        """
+        if time_to['total_minutes'] <= self.warn_threshold:
+            warn_message = str(summary) + " " + str(start) + " - " + str(end)
+            self.py3.notify_user(warn_message, 'warning', self.warn_timeout)
+
+    def _gstr_to_date(self, date_str):
+        """ Returns a dateime object from a google calendar date string."""
+        return datetime.datetime.strptime(date_str, '%Y-%m-%d')
+
+    def _gstr_to_datetime(self, date_time_str):
+        """ Returns a datetime object from a google calendar date/time string."""
+        tmp = ''.join(date_time_str.rsplit(':', 1))
+        return datetime.datetime.strptime(tmp, '%Y-%m-%dT%H:%M:%S%z')
+
+    def _datetime_to_str(self, date_time, dt_format):
+        """ Returns a strftime formatted string from a datetime object."""
+        return date_time.strftime(dt_format)
+
+    def _delta_time(self, date_time):
+        """
+        Returns in a dict the number of days/hours/minutes and total minutes
+        until date_time.
+        """
+        now = datetime.datetime.utcnow()
+
+        if date_time.tzinfo is not None \
+                and date_time.tzinfo.utcoffset(date_time) is not None:
+            now = now.replace(tzinfo=pytz.UTC)
+
+        diff = date_time - now
+
+        days = int(diff.days)
+        hours = int(diff.seconds / 3600)
+        minutes = int((diff.seconds / 60) - (hours * 60)) + 1
+        total_minutes = int((diff.seconds / 60) + (days * 24 * 60)) + 1
+
+        return {'days': days,
+                'hours': hours,
+                'minutes': minutes,
+                'total_minutes': total_minutes}
```

Comment:
(Suggestion) Place numbers above into this dict to omit few lines. It should look 95% same.

Reply:
Sorry, not sure what you mean. Which numbers?

Reply:
Ah. You need them outside. I didn't catch that. Whoops.